admin/packingslip.php

```diff
@@ -138,11 +138,54 @@
             <th class="dataTableHeadingContent">&nbsp;</th>
             <th class="dataTableHeadingContent" style="width: 70%"><?php echo TABLE_HEADING_PRODUCTS_NAME; ?></th>
             <th class="dataTableHeadingContent"><?php echo TABLE_HEADING_PRODUCTS_MODEL; ?></th>
+<?php
+          // -----
+          // Additional column-headings can be added.
+          //
+          // A watching observer can provide an associative array in the following format (for the products' listing ONLY):
+          //
+          // $extra_headings = array(
+          //     array(
+          //       'align' => $alignment,    // One of 'center', 'right', or 'left' (optional)
+          //       'text' => $value
+          //     ),
+          // );
+          //
+          // Observer notes:
+          // - Be sure to check that the $p2/$extra_headings value is specifically (bool)false before initializing, since
+          //   multiple observers might be injecting content!
+          // - If heading-columns are added, be sure to add the associated data columns, too, via the
+          //   'NOTIFY_ADMIN_PACKINGSLIP_DATA' notification.
+          //
+          $extra_headings = false;
+          $zco_notifier->notify('NOTIFY_ADMIN_PACKINGSLIP_HEADING', '', $extra_headings);
+          if (is_array($extra_headings)) {
+              foreach ($extra_headings as $heading_info) {
+                  $align = (isset($heading_info['align'])) ? (' text-' . $heading_info['align']) : '';
+?>
+            <th class="dataTableHeadingContent<?php echo $align; ?>"><?php echo $heading_info['text']; ?></th>
+<?php
+              }
+          }
+?>
           </tr>
         </thead>
         <tbody>
             <?php
-            for ($i = 0, $n = count($order->products); $i < $n; $i++) {
+            /*
+             * Notifier to allow packing slip to be sorted to required order
+             *
+
+             *              * Set $sort_order to the order->products array counter in the sequence you require the invoice to be displayed
+             */
+            $sort_order=false;
```

Comment:
https://www.php-fig.org/psr/psr-12/#:~:text=6.2.%20Binary%20operators,6.3.%20Ternary%20operators

Reply:
Sorry. Will correct. Thanks for pointing out.